pkg/clusterid/components.go

```diff
@@ -0,0 +1,48 @@
+package clusterid
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/pkg/core"
+	config_manager "github.com/kumahq/kuma/pkg/core/config/manager"
+	config_model "github.com/kumahq/kuma/pkg/core/resources/apis/system"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	core_runtime "github.com/kumahq/kuma/pkg/core/runtime"
+)
+
+func Setup(rt core_runtime.Runtime) error {
+	return rt.Add(&clusterID{rt: rt})
+}
+
+type clusterID struct {
+	rt core_runtime.Runtime
+}
+
+func (c *clusterID) Start(_ <-chan struct{}) error {
+	return createClusterID(c.rt)
+}
+
+func (c *clusterID) NeedLeaderElection() bool {
+	return true
+}
+
+func createClusterID(runtime core_runtime.Runtime) error {
+	resource := &config_model.ConfigResource{}
+
+	err := runtime.ConfigManager().Get(context.Background(), resource, store.GetByKey(config_manager.ClusterIdConfigKey, ""))
+	if err != nil {
+		if !store.IsResourceNotFound(err) {
+			return err
+		}
+		resource.Spec.Config = core.NewUUID()
+		if err := runtime.ConfigManager().Create(context.Background(), resource, store.CreateByKey(config_manager.ClusterIdConfigKey, "")); err != nil {
+			return errors.Wrap(err, "could not create config")
+		}
+	}
+	clusterId := resource.Spec.Config
+	runtime.SetClusterId(clusterId)
+
+	return nil
+}
```

Comment:
What do you think of putting this in `defaults`? I'd love to avoid having a separate package in pkg just for this.

Reply:
I did so, but then figured out `defaults` and `clusterid` work in different modes, `defaults` runs everywhere, `clusterid` runs only in Global and Standalone mode. 

I don't mind having `clusterid` as a separate component (even if it's tiny). I think we can create dir `pkg/components` and move all components here. Every component should have `func Setup(rt runtime.Runtime) error` function, own logger, documentation, and component's level tests. That's pretty much what we have today but it's not really ordered

Reply:
I don't know if you remember my presentation on one of the community call, but I also wanted to segment into a couple of main packages (util, core, api, components, plugins), where there are clear rules about dependencies between all packages. Components leave in separate components package. The main thing that I would like to see is to have 0 dependencies between components. They should be like separate processes that can be pulled off from the code.
https://docs.google.com/presentation/d/1EyEGpji6UHFYPyD4krORcIsD8aubDE9GXfDVlhhUNOo/edit?usp=sharing

but in terms of clusterid and defaults. Look that signing key is created also in global and standalone, but I put this in defaults. If you think about this, it's a matter of where you put the "if".
1) Put if whether to run small component
2) Put if in bigger component that creates defaults.
I prefer the second one, in this case, I think we "overpackage" many places in the codebase.
But I leave your judgment on which one to do, I'm not strongly attached to the second option.

Reply:
Yeah, of course, I remember your presentation. I totally get your idea about bigger components with less packaging. But unlike `signingKey` this `clusterid` doesn't really have "defaulting" functionality. It's more like "cluster init" functionality. So I'm not sure it belongs here.

Reply:
ClusterID is the same "default" as Signing Key which is a random value.